MapboxMobileEvents/MMELocationManager.m

```diff
@@ -1,39 +1,44 @@
 #import "MMELocationManager.h"
+#import "MMEBackgroundLocationServiceTimeoutHandler.h"
 #import "MMEUIApplicationWrapper.h"
 #import "MMEDependencyManager.h"
 #import "MMEEventsService.h"
 #import "MMEEventsConfiguration.h"
 #import <CoreLocation/CoreLocation.h>
 
-static const NSTimeInterval MMELocationManagerHibernationTimeout = 300.0;
-static const NSTimeInterval MMELocationManagerHibernationPollInterval = 5.0;
-
 const CLLocationDistance MMELocationManagerDistanceFilter = 5.0;
 const CLLocationDistance MMERadiusAccuracyMax = 300.0;
 
 NSString * const MMELocationManagerRegionIdentifier = @"MMELocationManagerRegionIdentifier.fence.center";
 
-@interface MMELocationManager () <CLLocationManagerDelegate>
+@interface MMELocationManager () <CLLocationManagerDelegate, MMEBackgroundLocationServiceTimeoutHandlerDelegate>
 
 @property (nonatomic) id<MMEUIApplicationWrapper> application;
 @property (nonatomic) CLLocationManager *locationManager;
 @property (nonatomic, getter=isUpdatingLocation, readwrite) BOOL updatingLocation;
-@property (nonatomic) NSDate *backgroundLocationServiceTimeoutAllowedDate;
-@property (nonatomic) NSTimer *backgroundLocationServiceTimeoutTimer;
+@property (nonatomic) MMEBackgroundLocationServiceTimeoutHandler *backgroundLocationServiceTimeoutTimerWrapper;
 @property (nonatomic) BOOL hostAppHasBackgroundCapability;
 @property (nonatomic) MMEEventsConfiguration *configuration;
 
 @end
 
 @implementation MMELocationManager
 
+- (void)dealloc {
+    _locationManager.delegate = nil;
```

Comment:
This is a great question. I had all the `stop...` variants in here, but since the location manager is being deallocated (and its delegate is going too) I'm assuming* that there's no need to call these. It's not clear from the documentation, but I think it's a fair assumption.

*uh oh.